bootstrap/secret/jwtprovider.go

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package secret
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces"
+	clientInterfaces "github.com/edgexfoundry/go-mod-core-contracts/v3/clients/interfaces"
+)
+
+type jwtSecretProvider struct {
+	secretProvider interfaces.SecretProvider
+}
+
+func NewJWTSecretProvider(secretProvider interfaces.SecretProvider) clientInterfaces.AuthenticationInjector {
+	return &jwtSecretProvider{
+		secretProvider: secretProvider,
+	}
+}
+
+func (self *jwtSecretProvider) AddAuthenticationData(req *http.Request) error {
+	if self.secretProvider == nil {
+		// Test cases or real code may invoke NewJWTSecretProvider(nil),
+		// though this is discouraged. In that case, just do nothing.
+		return nil
+	}
+
+	// Otherwise if there is a secret provider, get the JWT
+	jwt, err := self.secretProvider.GetSelfJWT()
+	if err != nil {
+		return err
+	}
+
+	// Only add authorization header if we can non-empty token back
```

Comment:
```suggestion
	// Only add authorization header if we get non-empty token back
```

Reply:
Fixed in https://github.com/edgexfoundry/go-mod-bootstrap/pull/404/commits/a31da98b6b01da966fd78ad2735f1782257a79d0